daemon/api.go

```diff
@@ -196,19 +196,30 @@ func loginUser(c *Command, r *http.Request) Response {
 	}
 
 	discharge, err := store.DischargeAuthCaveat(loginData.Username, loginData.Password, macaroon, loginData.Otp)
-	if err == store.ErrAuthenticationNeeds2fa {
+	switch err {
+	case store.ErrAuthenticationNeeds2fa:
 		twofactorRequiredResponse := &resp{
```

Comment:
can you get rid of the var here, and just return SyncResponse like the other block (at the 3rd of these *I* would be tempted to invent some kind of helper in response.go to make these kind of responses with kind and value but oh well for now)